app/forms/notification_preferences_form.rb

```diff
@@ -43,15 +41,15 @@ def categories
   def notification_categories(visible_categories)
     ui_notification_categories = []
 
-    enabled_categories.each do |category_name|
+    @notification_categories.enabled_categories.each do |category_name|
```

Comment:
It's a quite confusing, IMO, to have a method `notification_categories`, and also the variable `@notification_categories` that do not represent the same things.

Reply:
I didn't want to open this can of worms but the whole thing is very confusing and renaming to `available_categories` does not help IMHO.

We have a method `#categories`m then `#notification_categories`, then a variable `visible_categories` and then an instance variable `@available_categories`.

Does `available_categories` mean actually `all_categories`? Then what does `#categories` stand for and how does it differ from `#notification_categories`? `visible_categories` seems clear to me.

Reply:
Now I'm also forcing a particular order for categories in form:
`[account service application billing alert report]`